src/app/shared/types/course.model.ts

```diff
@@ -1,15 +1,22 @@
-import { Author } from "./author.model";
+import { Author } from './author.model';
 
 export interface Course {
   id: string;
   title: string;
   description: string;
-  creationDate: string;
+  creationDate?: string;
   duration: number;
   authors: string[];
 }
 
-export type CourseInfo = Omit<Course, "creationDate" | "authors"> & {
+export type CourseInfo = Omit<Course, 'creationDate' | 'authors'> & {
   date: string;
   authors: Author[];
 };
+
+export interface FilterCourse {
```

Comment:
Hmmm, I am not sure about the intent here, but as you query the the courses, the API will definitely return the courses with all object properties.

Reply:
this is not the response but the query param type. holding the filter arrays. According to swagger these are arrays.